pom.xml

```diff
@@ -363,6 +383,11 @@
       <artifactId>jsr305</artifactId>
       <version>3.0.2</version>
     </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>18.0</version>
+    </dependency>
```

Comment:
Most of this pull request makes sense except that we are trading one set of user problems with Guava 21 for another with Guava 18.  Could S3Proxy use the shaded jclouds Guava imports instead?  It looks like S3Proxy could mostly eliminate its use of Guava as well.

Reply:
Guava was added explicitly because the shade plugin doesn't pickup transitive dependencies. I went for 18.0 because that is the version jclouds-2.1.0 was pulling in. We could change this. 
S3Proxy will compile against this version and during the shading the classes will be updated to use the relocated guava classes. 
Any module using this will not see any guava dependency and can define their own.

Reply:
Thanks for explaining!  I looked at S3Proxy use of Guava and believe that we should excise it from the public interfaces at least.  I only found a few uses of `Optional` parameters which I replaced with `@Nullable` arguments in ec00aeb8cd23468366ed1a0eeaa9bae4f3bad6eb.

Reply:
@gaul sounds good. I think we still need the above otherwise the shading won't work, unless we wanted to move the shading and relocation up into jclouds itself...